### DISCUSSION TITLE
Car starts charging when plugged in

### DISCUSSION THREAD

**Discussion by githubrdonefun** [alandtse/tesla#371] [Q&A]
Hi,

I've set up an automation to increase my charge limit to a higher amount if electricity prices are lower and I've set a schedule to turn on charging at night and turn off in the morning.  I've also set it up to turn on preconditioning each weekday morning.  This all works great, but now that I'm using Home Assistant to handle the charging, the car automatically starts charging when I plug it in, no matter the time.  How do I prevent this and let it just charge during the hours I want?  I'd like to plug it in and it doesn't start charging like it did when I used the Tesla app to handle charging.  Any advice is appreciated.

**Comment by anhaff** [comment]
Probably a bunch of different solutions exists.

My choice is an automation which runs every 3 minutes when the car is at home and connected to the charger and performs one of two groups of tests: the first group if the car is currently charging and the second if it is not charging.

In the first group I check different reasons why the car actually should NOT charge and act accordingly - the automation turns charging off.

In the second group ... and so on 

Also mine starts charging when I insert the charger plug, but at the latest after 3 minutes it is turned off (or continues if there was no reason to turn it off).

**Comment by carleeno** [comment]
afaik there's not a way to disable charging by default on the car (unless you set a scheduled charge time, which it sounds like you're trying to avoid).

So a workaround would be to create a new automation triggered when car starts charging, condition it to only run when outside of your HA-defined charging schedule, and action it to turn off the charger switch. You probably want to also add some other conditions such as to ensure it only turns it off when at home.

Of course when you plug it in, it will still charge until HA detects that it started charging and turn it off, that would depend on your polling rate.

(As I finished typing this anhaff replied above with a very similar solution, I'll leave this as well as some may prefer the event-based automation vs a recurring automation)

**Reply by githubrdonefun** [reply]
Thanks, these are good ideas.  I like the automation idea instead of constant polling.  I'll try setting an automation that is triggers on the latch opening and the charging starting during the daytime periods.  I'll see how that goes.  I may eventually need to merge some of these automations.

**Comment by zipzagster** [comment]
I feel that manipulating charge limit is the way way to control charging. I Turn on off peak charging in the Tesla app and set the charge limit low. I have variable and usually cheap overnight electrical price. I set off peak charge to  5AM and 50%. I charge opportunistically to 90% when rates are at or below 0. When finished with an opportunistic charge set the charge limit back to 50%.

An HA automatic off function may have unintended consequences in unusual situation. I always want my car to be easily chargeable to a reasonable level like 40-50%. I also want the car charged to 50% minimum overnight regardless of pricing.

If anyone is a Comed hourly pricing customer I'm interested in discussing strategies for handling the price feed.

**Reply by githubrdonefun** [reply]
Oh cool I'm also using Comed hourly pricing.  I've been setting it to schedule only during certain hours and then if the price dips I bump my percentage charge up.

**Reply by zipzagster** [reply]
Unfortunately the HA Comed Integration does capture the time period which the price is referencing. It only captures the time the entity was updated. But its possible to attach that date/time as an attribute to the price. My strategy, which I'm still working on, is to predict a negative hour price and bump the charge limit.

I also use a zwave light sensor to signal HA when the charger LED starts and stops. I then run Force data update so that I accurately capture charging behavior.

**Reply by githubrdonefun** [reply]
I don't follow what you're saying, but I use their integration to turn other things on/off around my house based on if prices surge/drop.

**Reply by zipzagster** [reply]
The Comed API reports historical information, usually about five minutes late. Comed records the time period reported by the  millisUTC field:

[{"millisUTC":"1671732300000","price":"5.6"}

The home assistant comed integration doesn't import the millisUTC field. The time in the HA integration is simply the time HA recoded the information.

**Reply by githubrdonefun** [reply]
I stopped using the 5 minute price because I only care about the hourly price, which is what I get billed at.  The hourly price is just the average of the 5 minute price each hour, so I just use that.  For my purposes this works well.  Turning things on/off every 5 minutes if they are bouncing around your threshold is not ideal so I stopped doing that.

**Reply by zipzagster** [reply]
The comed price reporting system is broken at the moment. The price reported is currently over an hour old. You don't know that because the HA comed integration doesn't include the millisUTC field attached to the price.

**Reply by githubrdonefun** [reply]
It does sometimes have issues, but it’s working again.  I guess your point is that the api is unreliable but you’re not “interested in discussing strategies for handling the price feed” like you said. I’m not a developer but maybe if you are you could update the comed integration. The millisUTC field is available in the API. For what I need it currently works just fine. YMMV. Literally.

**Comment by anhaff** [comment]
Every day at 13:00 (or some minutes later) I receive new prices for the coming day - it is standard in Europe to consolidate the prices at that time. I know the battery capacity, my charging power and when I want the battery being full (at 80% or so) - unlike many others who want a full battery in the morning, I do not really care: I'm retired and usually do not leave for job in the morning.

Based on this, my algorithm chooses  the x cheapest hours of the y (y>x) hours with known prices to charge the car. If I really want to leave at some point of time with a 90% or more SoC, I limit the 'y' (cut it at the appropriate point of time. Graphically it looks like this (for quite an unusual time of leaving):
![image](https://user-images.githubusercontent.com/88614880/208930846-9edd0206-dbb3-43e4-a1ce-8ae48ba6020d.png)

**Reply by githubrdonefun** [reply]
Knowing the pricing the day ahead is interesting, but I can't do that.  Comed (Chicago, IL) has hourly pricing that is the wholesale price.  This changes in 5 minute intervals and I pay the hourly price average for each hour of the day.  https://hourlypricing.comed.com/hp-api/
